### ISSUE TITLE
Compute sortperm of a table

### ISSUE THREAD

**Issue by piever** [JuliaData/Tables.jl#49]
Using `Tables` one is quite close to be able to compute a `sortperm` of a `NamedTuple` of vectors.

In particular, let's say we have:

```julia
s = (a = [1, 2, 2, 1], b = 1:4)
t = Tables.rows(s)
```
Trying to call `sortperm(t)` "almost works" but with two obstacles:

1. `RowIterator` is not an `AbstractVector`. I actually don't understand why, because if `r = Tables.rows(s)` then `r[i]` is in my view well defined as `ColumnsRow(s, i)`.

2. There is no method `isless(::ColumnsRow, ::ColumnsRow)`. I'm a bit unsure what's the best way to define it efficiently. I'm afraid that relying on the standard `isless(::NamedTuple, ::NamedTuple)` is actually bad because it forces to materialize the `NamedTuple` and furthermore it converts to a `Tuple` which I think is bad in the case of missing data because of the type instability due to the covariance of tuples.

So, in short:

- Can `RowIterator` be made into a `AbstractVector`? Implementation-wise this is actually trivial
- Can we define an efficient `isless(::ColumnsRow, ::ColumnsRow)`? Implementation-wise this may be less trivial, esp. if we want to stay performant in the case of missing data

EDIT: opened a PR for the first part of the issue at #50

**Comment by quinnj** [comment]
Are we all good here now?

**Comment by quinnj** [comment]
Oh, sorry, I got my PRs mixed up; I thought we had merged the RowIterator one.
